src/main/kotlin/com.bugsnag.android.gradle/BugsnagUploadNdkTask.kt

```diff
@@ -0,0 +1,273 @@
+
+                // Output the file to a zip
+                val stdout = process.inputStream
+                outputZipFile(stdout, outputFile)
+                return if (process.waitFor() == 0) {
+                    outputFile
+                } else {
+                    logger.error("failed to generate symbols for " + arch + " see "
+                        + errorOutputFile.toString() + " for more details")
+                    null
+                }
+            } catch (e: Exception) {
+                logger.error("failed to generate symbols for " + arch + " " + e.message, e)
+            } finally {
+                outReader?.close()
+            }
+        } else {
+            logger.error("Unable to upload NDK symbols: Could not find objdump location for $arch")
+        }
+        return null
+    }
+
+    /**
+     * Uploads the given shared object mapping information
+     * @param mappingFile the file to upload
+     * @param arch the arch that is being uploaded
+     * @param sharedObjectName the original shared object name
+     */
+    private fun uploadSymbols(mappingFile: File, arch: String?, sharedObjectName: String?) {
+        // a SO file may not contain debug info. if that's the case then the mapping file should be very small,
+        // so we try and reject it here as otherwise the event-worker will reject it with a 400 status code.
+        if (!mappingFile.exists() || mappingFile.length() < VALID_SO_FILE_THRESHOLD) {
+            project.logger.warn("Skipping upload of empty/invalid mapping file: $mappingFile")
+            return
+        }
+        val mpEntity = MultipartEntity()
+        mpEntity.addPart("soSymbolFile", FileBody(mappingFile))
+        mpEntity.addPart("arch", StringBody(arch))
+        mpEntity.addPart("sharedObjectName", StringBody(sharedObjectName))
+        val bugsnag = project.extensions.getByType(BugsnagPluginExtension::class.java)
+        var projectRoot = bugsnag.projectRoot
+        if (projectRoot == null) {
+            projectRoot = projectDir.toString()
+        }
+        mpEntity.addPart("projectRoot", StringBody(projectRoot))
+        val request = BugsnagMultiPartUploadRequest()
+        request.variant = variant
+        request.variantOutput = variantOutput
+        request.uploadMultipartEntity(project, mpEntity, manifestInfoProvider.get())
+    }
+
+    /**
+     * Gets the path to the objdump executable to use to get symbols from a shared object
+     * @param arch The arch of the shared object
+     * @return The objdump executable, or null if not found
+     */
+    private fun getObjDumpExecutable(arch: String): File? {
+        try {
+            val override = getObjDumpOverride(arch)
+            val objDumpFile: File
+            objDumpFile = override?.let { File(it) } ?: findObjDump(project, arch)
+            check(!(!objDumpFile.exists() || !objDumpFile.canExecute())) { "Failed to find executable objdump at $objDumpFile" }
```

Comment:
I think this is a bit clearer

```suggestion
            check(objDumpFile.exists() && objDumpFile.canExecute())) { "Failed to find executable objdump at $objDumpFile" }
```